### DISCUSSION TITLE
MCUboot serial recovery through mcumgr not working

### DISCUSSION THREAD

**Discussion by aniruddhann** [zephyrproject-rtos/zephyr#40585] [Q&A] [answered]
Hi, 
I have flashed the MCUboot with Serial recovery mode enabled. I am trying to upload the firmware via mcumgr cli command. It seems the mcumgr "image upload" command works fine, however, after the upload is completed, the application is supposed to launch(which is not happening). 

It's also expected to show a hash value after I perform mcumgr "image list" command.(which is seen as unavailable)

To debug the flow, I had tried to route the MCUboot code debugging messages through other UART channel(other than the flashing/uploading channel/USART), but, could not see anything there. I interpret it that if serial recovery is to be used then none of other UART channel will work. (is it correct?)

I had also tried to debug it via RTT without success.
Please suggest, what's missing as such the application is not getting launched and any ways to debug/log the flow.
Thanks,
![mcumgr_error](https://user-images.githubusercontent.com/13480894/143008499-7238789f-487f-4fa4-b193-aafde64c8cc4.jpg)

**Comment by DerekSnell** [comment, marked as answer]
Hi @aniruddhann ,

Attached is a lab guide NXP uses to demonstrate MCUboot functioning with Zephyr.
[Zephyr_MCUboot_Lab_v1.0.pdf](https://github.com/zephyrproject-rtos/zephyr/files/7622139/Zephyr_MCUboot_Lab_v1.0.pdf)

Can you try replicating these steps?  Or post all the steps you are using to build, flash and update your hardware like in this guide?  That will help us understand any differences.

Best regards

**Reply by aniruddhann** [reply]
yes, it worked. Thanks a lot!
